neo/rawio/biocamrawio.py

```diff
@@ -0,0 +1,203 @@
+        t_stop = self._num_frames / self._sampling_rate
+        all_stops = [[t_stop]]
+        return all_stops[block_index][seg_index]
+
+    def _get_signal_size(self, block_index, seg_index, stream_index):
+        assert stream_index == 0
+        return self._num_frames
+
+    def _get_signal_t_start(self, block_index, seg_index, stream_index):
+        assert stream_index == 0
+        return self._segment_t_start(block_index, seg_index)
+
+    def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop,
+                                stream_index, channel_indexes):
+        if i_start is None:
+            i_start = 0
+        if i_stop is None:
+            i_stop = self._num_frames
+
+        data = self._read_function(self._rf, i_start, i_stop, self._num_chanels)
+        # transform to slice if possible
+        if sorted(channel_indexes) == channel_indexes and np.all(np.diff(channel_indexes) == 1):
+            channel_ids = slice(channel_indexes[0], channel_indexes[0] + len(channel_indexes))
+        return data[:, channel_indexes]
+
+
+def openBiocamFile(filename, mea_pitch, verbose=False):
+    """Open a Biocam hdf5 file, read and return the recording info, pick te correct method to access raw data,
+    and return this to the caller."""
+    try:
+        import h5py
+        HAVE_H5PY = True
+    except ImportError:
+        HAVE_H5PY = False
+    assert HAVE_H5PY, 'h5py is not installed'
+
+    rf = h5py.File(filename, 'r')
+    # Read recording variables
+    recVars = rf.require_group('3BRecInfo/3BRecVars/')
+    # bitDepth = recVars['BitDepth'].value[0]
+    # maxV = recVars['MaxVolt'].value[0]
+    # minV = recVars['MinVolt'].value[0]
+    nFrames = recVars['NRecFrames'][0]
+    samplingRate = recVars['SamplingRate'][0]
+    signalInv = recVars['SignalInversion'][0]
+    # Read chip variables
+    chipVars = rf.require_group('3BRecInfo/3BMeaChip/')
+    nCols = chipVars['NCols'][0]
+    # Get the actual number of channels used in the recording
+    file_format = rf['3BData'].attrs.get('Version')
+    if file_format == 100:
+        nRecCh = len(rf['3BData/Raw'][0])
+    elif (file_format == 101) or (file_format == 102):
+        nRecCh = int(1. * rf['3BData/Raw'].shape[0] / nFrames)
+    else:
+        raise Exception('Unknown data file format.')
+
+    if verbose:
+        print('# 3Brain data format:', file_format, 'signal inversion', signalInv)
+        print('#       signal range: ', recVars['MinVolt'][0], '- ', recVars['MaxVolt'][0])
+        print('# channels: ', nRecCh)
+        print('# frames: ', nFrames)
+        print('# sampling rate: ', samplingRate)
+    # get channel locations
+    r = (rf['3BRecInfo/3BMeaStreams/Raw/Chs'][()]['Row'] - 1) * mea_pitch
+    c = (rf['3BRecInfo/3BMeaStreams/Raw/Chs'][()]['Col'] - 1) * mea_pitch
+    rawIndices = np.vstack((r, c)).T
+    # assign channel numbers
+    chIndices = np.array([(x - 1) + (y - 1) * nCols for (y, x) in rawIndices])
```

Comment:
For consistency I would remove all camel case variable name.

Reply:
done